gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/language/grammar/TraversalPredicateVisitorTest.java

```diff
@@ -90,6 +90,9 @@
                 {"TextP.endingWith('hakuna')", TextP.endingWith("hakuna")},
                 {"TextP.notEndingWith('hakuna')", TextP.notEndingWith("hakuna")},
                 {"TextP.notStartingWith('hakuna')", TextP.notStartingWith("hakuna")},
+                {"TextP.regex('^h')", TextP.regex("^h")},
```

Comment:
Please add the following tests:
1. a pattern starting with `~` because this is a reserved keyword in Gremlin and we want to ensure the implementation works correctly.
2. a pattern containing the Java implementation specific flag such as case-insensitive search `(?i)` to demonstrate that our regex works with Java specific flags.